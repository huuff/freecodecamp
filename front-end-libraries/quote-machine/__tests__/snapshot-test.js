import React from 'react'
import ReactDOM from 'react-dom'

import Enzyme, {shallow, render, mount} from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import QuoteBox from '../src/quote-box'

Enzyme.configure({adapter: new Adapter()})

it('QuoteBox renders corrently', () => {
  const wrapper = shallow(<QuoteBox
    quote={{ text: "", author: "", tags: []}}

    />)
  expect(toJson(wrapper)).toMatchSnapshot()
})
