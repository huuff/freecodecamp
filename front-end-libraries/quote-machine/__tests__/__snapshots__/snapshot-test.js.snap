// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuoteBox renders corrently 1`] = `
<div
  className="row min-vh-100 justify-content-center align-items-center"
  id="quote-box"
>
  <div
    className="card shadow w-50 mb-3"
  >
    <div
      className="card-header"
    >
      <h4>
        Quote of the moment
      </h4>
    </div>
    <div
      className="card-body"
    >
      <CSSTransition
        appear={true}
        classNames="fade-effect"
        timeout={1000}
      >
        <div
          className="d-flex justify-content-center my-5"
        >
          <div
            className="spinner-border text-primary"
            style={
              Object {
                "height": "3rem",
                "width": "3rem",
              }
            }
          />
        </div>
      </CSSTransition>
    </div>
    <div
      className="card-footer"
    >
      <Tweet
        author=""
        quote=""
      />
      <button
        className="btn btn-primary float-end ms-2"
        id="author-quote"
        onClick={[Function]}
      >
        Quote from this author
      </button>
      <button
        className="btn btn-primary float-end ms-2"
        id="new-quote"
        onClick={[Function]}
      >
        Random quote
      </button>
    </div>
    <div />
  </div>
</div>
`;
